src/More/About.spec.tsx

```diff
@@ -1,44 +1,46 @@
 import React from "react"
 import { Linking } from "react-native"
-import { render, fireEvent } from "@testing-library/react-native"
-import {
-  getApplicationName,
-  getVersion,
-  getBuildNumber,
-} from "react-native-device-info"
+import { render, fireEvent, wait } from "@testing-library/react-native"
 
 import AboutScreen from "./About"
-
-jest.mock("react-native-device-info", () => {
-  return {
-    getApplicationName: jest.fn(),
-    getBuildNumber: jest.fn(),
-    getVersion: jest.fn(),
-  }
-})
+import * as NativeModule from "../gaen/nativeModule"
 
 describe("About", () => {
-  it("shows the name of the application", () => {
+  it("shows the name of the application", async () => {
     const applicationName = "application name"
 
-    ;(getApplicationName as jest.Mock).mockReturnValueOnce(applicationName)
+    const applicationNameSpy = jest
+      .spyOn(NativeModule, "getApplicationName")
+      .mockReturnValue(Promise.resolve(applicationName))
 
     const { getByText } = render(<AboutScreen />)
 
-    expect(getByText(applicationName)).toBeDefined()
+    await wait(() => {
+      expect(applicationNameSpy).toHaveBeenCalled()
+      expect(getByText(applicationName)).toBeDefined()
+    })
   })
 
-  it("shows the build and version number of the application", () => {
-    const buildNumber = 8
-    const versionNumber = 0.18
+  it("shows the build and version number of the application", async () => {
+    const buildNumber = "8"
+    const versionNumber = "0.18"
+    const fullString = `${versionNumber} (${buildNumber})`
 
-    ;(getBuildNumber as jest.Mock).mockReturnValueOnce(buildNumber)
-    ;(getVersion as jest.Mock).mockReturnValueOnce(versionNumber)
+    const buildNumberSpy = jest
+      .spyOn(NativeModule, "getBuildNumber")
+      .mockReturnValue(Promise.resolve(buildNumber))
```

Comment:
```suggestion
      .mockResolvedValueOnce(buildNumber)
```

Reply:
the method returns a `Promise<string>` here; it registers an error when passed a `string`